extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -125,6 +137,17 @@ private static void validateBuildTimeConfig(AggregatedDataSourceBuildTimeConfigB
         String fullDataSourceName = aggregatedConfig.isDefault() ? "default datasource"
                 : "datasource named '" + aggregatedConfig.getName() + "'";
 
+        if (jdbcBuildTimeConfig.tracing) {
+            try {
+                Class.forName(DataSources.TRACING_DRIVER_CLASSNAME, true, Thread.currentThread().getContextClassLoader());
```

Comment:
You could use `QuarkusClassLoader.isClassPresentAtRuntime(DataSources.TRACING_DRIVER_CLASSNAME)` here too. It's more correct anyway as the class could be present in the deployment classpath but not the runtime one.